Lib/fontTools/varLib/errors.py

```diff
@@ -32,32 +32,72 @@ def __init__(self, merger, args):
+        elif hasattr(ttf.reader, "file") and hasattr(ttf.reader.file, "name"):
+            return ttf.reader.file.name
+        else:
+            return "master number %i" % ix
+
+    def _offender(self, cause):
+        reason = cause["reason"].value
+        if "expected" in cause and "got" in cause:
+            index = [x == cause["expected"] for x in cause["got"]].index(False)
+            return index, self._master_name(index)
+        if reason == VarLibMergeFailure.FoundANone:
+            index = [x is None for x in cause["got"]].index(True)
+            return index, self._master_name(index)
+        return None, None
+
+    def _incompatible_features(self, offender_index):
         cause, stack = self.args[0], self.args[1:]
-        fontnames = [
-            ttf["name"].getDebugName(1) + " " + ttf["name"].getDebugName(2)
-            for ttf in self.merger.ttfs
+        bad_ttf = self.merger.ttfs[offender_index]
+        good_ttf = self.merger.ttfs[offender_index - 1]
+
+        good_features = [
+            x.FeatureTag for x in good_ttf[stack[-1]].table.FeatureList.FeatureRecord
         ]
+        bad_features = [
+            x.FeatureTag for x in bad_ttf[stack[-1]].table.FeatureList.FeatureRecord
+        ]
+        return (
+            "\nIncompatible features between masters.\n"
+            f"Expected: {', '.join(good_features)}.\n"
+            f"Got: {', '.join(bad_features)}.\n"
+        )
+
+    def __str__(self):
+        cause, stack = self.args[0], self.args[1:]
         context = "".join(reversed(stack))
         details = ""
         reason = cause["reason"].value
-        if reason == VarLibMergeFailure.FoundANone:
-            offender = [x is None for x in cause["got"]].index(True)
-            details = (
-                f"\n\nThe problem is likely to be in {fontnames[offender]}:\n"
-                f"{stack[0]}=={cause['got']}\n"
-            )
+        offender_index, offender = self._offender(cause)
+        if offender:
+            details = f"\n\nThe problem is likely to be in {offender}:\n"
+        if cause["reason"] == VarLibMergeFailure.FoundANone:
+            details = details + f"{stack[0]}=={cause['got']}\n"
+        elif (
+            cause["reason"] == VarLibMergeFailure.ShouldBeConstant
+            and stack[0] == ".FeatureCount"
+        ):
+            # Common case
+            details = details + self._incompatible_features(offender_index)
         elif "expected" in cause and "got" in cause:
             offender = [x == cause["expected"] for x in cause["got"]].index(False)
             got = cause["got"][offender]
-            details = (
-                f"\n\nThe problem is likely to be in {fontnames[offender]}:\n"
+            details = details + (
                 f"Expected to see {stack[0]}=={cause['expected']}, instead saw {got}\n"
             )
 
         if (
-            reason == VarLibMergeFailure.UnsupportedFormat
```

Comment:
Maybe compare by identity? That's what the docs recommend: https://docs.python.org/3/library/enum.html?highlight=enum#comparisons